src/index.ts

```diff
@@ -5,13 +5,48 @@ type PineconeClientConfiguration = {
   apiKey: string
 }
 
+Error.prepareStackTrace = () => '';
+
+async function streamToArrayBuffer(stream: ReadableStream<Uint8Array>): Promise<Uint8Array> {
+  let result = new Uint8Array(0);
+  const reader = stream.getReader();
+  while (true) { // eslint-disable-line no-constant-condition
+    const { done, value } = await reader.read();
+    if (done) {
+      break;
+    }
+
+    const newResult = new Uint8Array(result.length + value.length);
+    newResult.set(result);
+    newResult.set(value, result.length);
+    result = newResult;
+  }
+  return result;
+}
+
 async function handler(func: Function, args: any) {
   try {
     return await func(args)
   } catch (error) {
     // @ts-ignore
-    const message = error?.response?.data?.message || null
-    throw `PineconeClient: Error calling ${func.name.replace("bound ", "")}: ${error} ${message ? `(${message})` : ''}`
+    if (error && error.response) {
+      // @ts-ignore
```

Comment:
Can we use some sort of type guard to reduce the need for the ts-ignores here?
Like would something like

`if (error instanceof ResponseError) {...}` work?